tests/testsuite/check.rs

```diff
@@ -1176,3 +1176,270 @@ fn git_manifest_with_project() {
         )
         .run();
 }
+
+#[cargo_test]
+fn check_fixable_warning() {
+    let foo = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+            "#,
+        )
+        .file("src/lib.rs", "use std::io;")
+        .build();
+
+    foo.cargo("check")
+        .masquerade_as_nightly_cargo(&["auto-fix note"])
+        .with_status(0)
+        .with_stderr_contains("[..] (run `cargo fix --lib -p foo` to apply 1 suggestion)")
+        .run();
+}
+
+#[cargo_test]
+fn check_fixable_not_nightly() {
+    let foo = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+            "#,
+        )
+        .file("src/lib.rs", "use std::io;")
+        .build();
+
+    foo.cargo("check")
+        .with_status(0)
+        .with_stderr_does_not_contain("[..] (run `cargo fix --lib -p foo` to apply 1 suggestion)")
```

Comment:
I have a bit concern of overmatching the string here. We might forget to update this "not_contain" when changing the error message. Then it will become always true. Maybe matching less or just matching the entire stderr? 
(it might be overkill tho)